src/test/java/uk/ac/ebi/eva/pipeline/io/readers/VcfReaderTest.java

```diff
@@ -0,0 +1,64 @@
+package uk.ac.ebi.eva.pipeline.io.readers;
+
+import org.junit.Test;
+import org.opencb.biodata.models.variant.Variant;
+import org.opencb.biodata.models.variant.VariantSource;
+import org.opencb.biodata.models.variant.VariantStudy;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.test.MetaDataInstanceFactory;
+import uk.ac.ebi.eva.test.utils.JobTestUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import static org.junit.Assert.*;
+
+/**
+ * {@link VcfReader}
+ * input: a Vcf file
+ * output: a VariantAnnotation each time its `.read()` is called
+ *
+ * incorrect input lines should not make the reader fail.
+ *
+ * @author Jose Miguel Mut Lopez &lt;jmmut@ebi.ac.uk&gt;
+ */
+public class VcfReaderTest {
```

Comment:
Some tests missing:

* Incorrect VCF file
* Testing that compressed and non-compressed VCF files are readable